index.d.ts

```diff
@@ -0,0 +1,355 @@
+    prompts?: string[];
+    refreshTokenRotation?: "rotateAndConsume" | "none";
+    renderError?: (ctx: Koa.Context, out: object, error: Error) => void;
+    responseTypes?: string[];
+    revocationEndpointAuthMethods?: AuthMethod[];
+    routes?: IRoutes;
+    scopes?: string[];
+    subjectTypes?: SubjectType;
+    tokenEndpointAuthMethods?: AuthMethod[];
+    ttl?: {
+      AccessToken?: number;
+      AuthorizationCode?: number;
+      ClientCredentials?: number;
+      DeviceCode?: number;
+      IdToken?: number;
+      RefreshToken?: number;
+    };
+    uniqueness?: (ctx: Koa.Context, jti: string, expiresAt: number) => boolean;
+    userCodeConfirmSource?: (
+      ctx: Koa.Context,
+      form: string,
+      client: object,
+      deviceInfo: object
+    ) => void;
+    userCodeInputSource?: (ctx: Koa.Context, form: string, out: object, err: Error) => void;
+    whitelistedJWA?: {
+      authorizationEncryptionAlgValues?: EncryptionAlgValue[];
+      authorizationEncryptionEncValues?: EncryptionEncValue[];
+      authorizationSigningAlgValues?: SigningAlgValue[];
+      idTokenEncryptionAlgValues?: EncryptionAlgValue[];
+      idTokenEncryptionEncValues?: EncryptionEncValue[];
+      idTokenSigningAlgValues?: SigningAlgValue[];
+      introspectionEncryptionAlgValues?: EncryptionAlgValue[];
+      introspectionEncryptionEncValues?: EncryptionEncValue[];
+      introspectionEndpointAuthSigningAlgValues?: SigningAlgValue[];
+      introspectionSigningAlgValues?: SigningAlgValue[];
+      requestObjectEncryptionAlgValues?: EncryptionAlgValue[];
+      requestObjectEncryptionEncValues?: EncryptionEncValue[];
+      requestObjectSigningAlgValues?: SigningAlgValue[];
+      revocationEndpointAuthSigningAlgValues?: SigningAlgValue[];
+      tokenEndpointAuthSigningAlgValues?: SigningAlgValue[];
+      userinfoEncryptionAlgValues?: EncryptionAlgValue[];
+      userinfoEncryptionEncValues?: EncryptionEncValue[];
+      userinfoSigningAlgValues?: SigningAlgValue[];
+    };
+  }
+
+  class Provider extends EventEmitter {
+    app: Koa;
+
+    domain: {};
+
+    proxy: boolean;
+
+    constructor(url: string, config?: IConfiguration);
+
+    initialize(config: {
+      adapter?: new (name: string) => IAdapter;
+      clients?: IClient[];
+      keystore?: unknown;
+    }): Promise<this>;
+
+    interactionDetails(ctx: IncomingMessage): Promise<ISession>;
+
+    setProviderSession(req: IncomingMessage, res: ServerResponse, {}): Promise<ISession>;
+
+    interactionFinished(req: IncomingMessage, res: ServerResponse, {}): Promise<void>;
+
+    callback: Handler;
+    listen(port: string | number): void;
+
+    Client: IClient;
+
+    session: ISession;
+
+    params: {
+      response_type: "none";
+    };
+    result: boolean;
+  }
+
+  export const createKeyStore: JWK.createKeyStore;
+  export const asKeyStore: JWK.asKeyStore;
+  export const asKey: JWK.asKey;
+
+  export default Provider;
```

Comment:
as a minimal type implementation let's type the exported errors property

see:
- https://github.com/panva/node-oidc-provider/blob/v5.1.2/test/provider/provider_class.test.js#L12-L41
- https://github.com/panva/node-oidc-provider/blob/v5.1.2/lib/helpers/errors.js